doc/panama_ffi.md

```diff
@@ -47,6 +46,63 @@ Of course, since accessing the entire native heap is inherently *unsafe*, access
 
 `MemoryAddress`, like `MemorySegment` , implements the `Addressable` interface, which is a functional interface whose method projects an entity down to a `MemoryAddress` instance. In the case of `MemoryAddress` such a projection is the identity function; in the case of a memory segment, the projection returns the `MemoryAddres` instance for the segment's base address. This abstraction allows to pass either memory address or memory segments where an address is expected (this is especially useful when generating native bindings).
 
+### Segment allocators
+
+Idiomatic C code implicitly relies on stack allocation to allow for concise variable declarations; consider this example:
+
+```c
+void foo(int i, int *size);
+
+int size = 5;
+foo(42, &size);
+```
+
+Here the function `foo` takes an output parameter, a pointer to an `int` variable. Unfortunately (and we have seen part of that pain in the `qsort` example above), implementing this idiom in Panama is not straightforward:
+
+```java
+MethodHandle foo = ...
+MemorySegment size = MemorySegment.allocateNative(C_INT);
+MemoryAccess.setInt(size, 5);
+foo.invokeExact(42, size);
```

Comment:
I think this should call `size.address()` to be equivalent with the C code. Also, reachability of the segment is problematic in that case, so maybe this should switch to using an explicit resource scope.

Reply:
I've rewritten this section as I noted that it was referring to native method handles, which were not yet discussed. Instead I focussed on "creating a native array" as a way to demonstrate "clunkiness" of base memory access API.

Reply:
As for deterministic deallocation, I've changed all samples to just rely on it - and in the very first case (e.g. call to strlen) I explain that going from a segment to an address is a bit tricky.